### ISSUE TITLE
How to convert instance-level masks to semantic masks in the CRAG dataset?

### ISSUE THREAD

**Issue by li-qi-lin** [cvlab-stonybrook/SAMPath#23]
I am currently working with the **CRAG dataset**, which provides **instance-level segmentation masks** of adenocarcinoma and benign glands in colon cancer.

I would like to know **how to convert these instance-level masks to **semantic masks** in your experiment**.

Could you provide some guidance or suggestions on how to achieve this?

Thank you！

**Comment by jingweizhang-xyz** [comment]
Simply regard all instances as one class: gland and regard the rest region as background.

**Comment by li-qi-lin** [comment]
Thank you！But I still have some questions.

1、The paper of CRAG mentions both **healthy and malignant glands**, but it isn’t entirely clear if there’s a need to further differentiate these regions within each gland.

2、If additional differentiation within glands is recommended, could you please provide guidance or any established criteria for such annotations?

Understanding this distinction would be very helpful for correctly interpreting and using the mask labels in the dataset. Thank you for your assistance!

**Comment by jingweizhang-xyz** [comment]
1. We did not try to distinguish between healthy and malignant. If you have such labels, you can have a try.
2. I do not know about such criteria or annotations.

**Comment by dw1360585641** [comment]
> 只需将所有实例视为一个类：gland，并将其余区域视为背景。
那么应该这样设置参数吗"dataset": {
        "dataset_root": "/data/kb/wdw/sampath/SAMPath-main/kaggle",
        "dataset_csv_path": "/data/kb/wdw/sampath/SAMPath-main/kaggle/200.csv",
        "val_fold_id": 0,
        "num_classes": 3,

        "ignored_classes": (0),
        "ignored_classes_metric": None,# if we do not count background, set to 1 (bg class)
        "image_hw": (1024, 1024), # default is 1024, 1024

        "feature_input": False, # or "True" for *.pt features
        "dataset_mean": (0.485, 0.456, 0.406),
        "dataset_std": (0.229, 0.224, 0.225),

**Comment by jingweizhang-xyz** [comment]
Looks ok, you can have a try.

**Comment by dw1360585641** [comment]
> 看起来不错，你可以试试。

1. i have a question about dataset,like this :  in my dataset, all the image  has two class: background , glomerulus，and i set background（0），glomerulus（1）， so the follow is right ？    
   “val_fold_id”： 0，
    “num_classes”： 3，
    "ignored_classes": None,
    "ignored_classes_metric": None,
2. Besides，the number of fold in the file（.csv），i know img_id is filepath, fold = -1 is test-image, fold =0 is val-image, but fold =1,fold =2,fold=3 mean to what in the pixel of mask ?  Because if i set all  fold =1 execept filename with fold =-1,0 and the result is wrong， but success if i set fold =2 ，why？
my .csv like this：
img_id,fold
image0,-1
image1,-1
image2,-1
image3,-1
image4,-1
image5,-1
image6,-1
image7,-1
image8,-1
image9,-1
image10,0
image11,0
image12,0
image13,0
image14,0
image15,0
image16,0
image17,0
image18,0
image19,0
image20,0
image21,0
image22,0
image23,0
image24,0
image25,0
image26,0
image27,0
image28,0
image29,0
image30,0
image31,0
image32,0
image33,0
image34,0
image35,0
image36,0
image37,0
image38,0
image39,2
image40,2
image41,2
image42,2
image43,2
image44,2
image45,2
image46,2
image47,2
image48,2
image49,2
image50,2
image51,2
image52,2
image53,2
image54,2
image55,2
image56,2
image57,2
image58,2
image59,2
image60,2
image61,2
image62,2
image63,2

**Comment by jingweizhang-xyz** [comment]
For 1, it should be correct, we use an extra class in this case.
For 2, fold==-1 means this sample is in the test set, "val_fold_id" in the cfg file defines the validation set. E.g. default, val_fold_id is 0. All samples with fold==0 are the validation set and all other folds are the training set.

**Comment by dw1360585641** [comment]
so in the trainset，the fold id how to make？
for exemple，my trainset have 2claas ，background and cell，so fold-id for trainset is what ？



---Original---
From: "Jingwei ***@***.***&gt;
Date: Tue, Dec 3, 2024 06:14 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [cvlab-stonybrook/SAMPath] How to convert instance-level masks tosemantic masks in the CRAG dataset? (Issue #23)




 
For 1, it should be correct, we use an extra class in this case.
 For 2, fold==-1 means this sample is in the test set, "val_fold_id" in the cfg file defines the validation set. E.g. default, val_fold_id is 0. All samples with fold==0 are the validation set and all other folds are the training set.

**Comment by jingweizhang-xyz** [comment]
You can set it to 1 for simplicity. But we assume that you have a validation dataset which fold-id is 0. Not sure if it can run without a validation set. You may need to change [image_mask_dataset.py](https://github.com/cvlab-stonybrook/SAMPath/blob/main/image_mask_dataset.py)
